src/analysis/retail/warlock/affliction/modules/spells/SiphonLife.tsx

```diff
@@ -1,78 +1,58 @@
-  static dependencies = {
-    enemies: Enemies,
-  };
-  protected enemies!: Enemies;
+const SL_DAMAGE_BONUS = 0.2;
 
-  get uptime() {
-    return this.enemies.getBuffUptime(TALENTS.SIPHON_LIFE_TALENT.id) / this.owner.fightDuration;
+class SiphonLife extends Analyzer {
+  get dps() {
+    return (this.bonusDmg / this.owner.fightDuration) * 1000;
   }
 
-  get suggestionThresholds() {
-    return {
-      actual: this.uptime,
-      isLessThan: {
-        minor: 0.95,
-        average: 0.9,
-        major: 0.8,
-      },
-      style: ThresholdStyle.PERCENTAGE,
-    };
-  }
+  bonusDmg = 0;
 
   constructor(options: Options) {
     super(options);
     this.active = this.selectedCombatant.hasTalent(TALENTS.SIPHON_LIFE_TALENT);
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.CORRUPTION_DEBUFF),
+      this.onCorruptionDamage,
+    );
   }
 
-  suggestions(when: When) {
-    when(this.suggestionThresholds).addSuggestion((suggest, actual, recommended) =>
-      suggest(
-        <>
-          Your <SpellLink spell={TALENTS.SIPHON_LIFE_TALENT} /> uptime can be improved. Try to pay
-          more attention to your Siphon Life on the boss, perhaps use some debuff tracker.
-        </>,
-      )
-        .icon(TALENTS.SIPHON_LIFE_TALENT.icon)
-        .actual(
-          defineMessage({
-            id: 'warlock.affliction.suggestions.siphonLife.uptime',
-            message: `${formatPercentage(actual)}% Siphon Life uptime`,
-          }),
-        )
-        .recommended(`>${formatPercentage(recommended)}% is recommended`),
-    );
+  onCorruptionDamage(event: DamageEvent) {
+    this.bonusDmg += calculateEffectiveDamage(event, SL_DAMAGE_BONUS);
   }
 
-  subStatistic() {
-    const history = this.enemies.getDebuffHistory(TALENTS.SIPHON_LIFE_TALENT.id);
+  statistic() {
     return (
-      <div className="flex">
-        <div className="flex-sub icon">
-          <SpellIcon spell={TALENTS.SIPHON_LIFE_TALENT} />
-        </div>
-        <div className="flex-sub value" style={{ width: 140 }}>
-          {formatPercentage(this.uptime, 0)} % <small>uptime</small>
-        </div>
-        <div className="flex-main chart" style={{ padding: 15 }}>
-          <UptimeBar
-            uptimeHistory={history}
-            start={this.owner.fight.start_time}
-            end={this.owner.fight.end_time}
-          />
-        </div>
-      </div>
+      <Statistic
+        category={STATISTIC_CATEGORY.TALENTS}
+        size="flexible"
+        tooltip={
+          <>
+            {formatThousands(this.bonusDmg)} bonus damage
+            <br />
+            <br />
+            test
+          </>
+        }
+      >
+        <BoringSpellValueText spell={TALENTS.SIPHON_LIFE_TALENT}>
+          {formatNumber(this.dps)} DPS{' '}
+          <small>
+            {formatPercentage(this.owner.getPercentageOfTotalDamageDone(this.bonusDmg))} % of total
+          </small>
+        </BoringSpellValueText>
```

Comment:
You should consider using `<ItemDamageDone/>` component for displaying throughput information.
Looks like this:
![image](https://github.com/user-attachments/assets/28962d61-4c52-4dfe-b6de-9bb8fb6aa04e)

```suggestion
       <BoringSpellValueText spell={TALENTS.SIPHON_LIFE_TALENT}>
          <ItemDamageDone amount={this.bonusDmg} />
        </BoringSpellValueText>
```

Reply:
I'm gonna be honest here. I copied the AbsoluteCorruption.tsx file and changed the relevant values to Siphon Life's Values where possible since those two talents are literally almost the exact same thing. So how would I implement your suggestion?

Reply:
You should be able to import the component as such
`import ItemDamageDone from 'parser/ui/ItemDamageDone';`
And then parse `this.bonusDmg` as the `amount` param.
`<ItemDamageDone amount={this.bonusDmg} />`

Reply:
Thanks, I have now implemented the above code and tidied up any errors caused by it. I also implemented this code into AbsoluteCorruption.tsx to maintain consistency between the two.